File: frontend/src/service/product.service.js
import axios from "axios";

const API_URL = "http://localhost:8080";

class ProductService {
    
    saveProduct(product) {
        return axios.post(API_URL + "/saveProduct", product);
    }

    getAllProducts() {
        return axios.get(API_URL + "/getAll");
    }

    getProductById(id) {
        return axios.get(API_URL + "/getById/" + id);
    }

    deleteProductById(id) {
        return axios.delete(API_URL + "/deleteProduct/" + id);
    }

    editProduct(product, id) {
        return axios.put(API_URL + "/editProduct/" + id, product);
    }
}

export default new ProductService();

File: src/main/java/com/PMS/PMS/service/ProductService.java
package com.PMS.PMS.service;

import com.PMS.PMS.model.Product;

import java.util.List;

public interface ProductService {
    public Product saveProduct(Product product);

    public List<Product> getAllProducts();

    public Product getProductById(Integer id);

    public String deleteProduct(Integer id);

    public Product editProduct(Product product,Integer id);
}
